src/CombFilter/CombFilter.h

```diff
@@ -0,0 +1,52 @@
+#ifndef __CombFilter_hdr__
+#define __CombFilter_hdr__
+#include "ErrorDef.h"
+#include "RingBuffer.h"
+
+class CCombFilterBase
+{
+public:
+	CCombFilterBase(int, int);
+	~CCombFilterBase();
+	Error_t setDelayLength(int iDelayInSample)
+	{
+		m_iDelayLength = iDelayInSample;
+		return Error_t::kNoError;
+	}
+	Error_t setGain(float g)
+	{
+		m_fGain = g;
+		return Error_t::kNoError;
+	}
+	int getDelayLength()
+	{
+		return m_iDelayLength;
+	}
+	float getGainValue()
+	{
+		return m_fGain;
+	}
+	virtual Error_t combFilter(float** ppfInputBuffer, float** ppfOutputBuffer, int iNumberOfFrames) = 0;
+protected:
+	int m_iDelayLength;
+	float m_fGain;
+	int m_iNumChannels;
+	CRingBuffer<float>** pCRingBuff = 0;
+
+};
+
+class CCombFIR : public CCombFilterBase
+{
+public:
+	using CCombFilterBase::CCombFilterBase;
```

Comment:
Compiler does not like "semantic issue
<img width="816" alt="Screen Shot 2022-02-16 at 7 33 17 PM" src="https://user-images.githubusercontent.com/69863091/154381219-b62e5c6f-69f0-47c2-948b-96d3d6a2cb15.png">
"

Reply:
This works fine for me on windows with Visual Studio 2019 with C++14 standard. Could you provide your environment and compiler version? If this causes cross-platform issue then I would definitely change it! Thank you!

Reply:
https://en.cppreference.com/w/Template:cpp/compiler_support/11
In line "Inheriting Constructors" it shows Apple Clang supports this feature, This seemed off to me.

Reply:
That is off, it is probably just a mistake on my end for some reason. I had issues with constructors when talking to Jason as well.